src/celutil/utf8.cpp

```diff
@@ -890,3 +902,93 @@ ReplaceGreekLetterAbbr(char *dst, unsigned int dstSize, const char* src, unsigne
 
     return 0;
 }
+
+static int findGreekNameIndexBySubstr(const std::string &, int = 0, unsigned int = UINT_MAX);
+static std::string firstGreekAbbrCompletion(const std::string &);
+
+bool inline isSubstringIgnoringCase(const std::string &s0, const std::string &s1, size_t n)
+{
+    return UTF8StringCompare(s0, s1, n, true) == 0;
+}
+
+static int findGreekNameIndexBySubstr(const std::string &s, int start, unsigned int n)
+{
+    Greek *instance = Greek::getInstance();
+
+    if (s.empty())
+        return -1;
+
+    for (int i = start; i < instance->nLetters; i++)
+    {
+        if (isSubstringIgnoringCase(instance->names[i], s, n))
+            return i;
+    }
+
+    for (int i = start; i < instance->nLetters; i++)
+    {
+        if (isSubstringIgnoringCase(instance->abbrevs[i], s, n))
+            return i;
+    }
+
+    return -1;
+}
+
+static std::string firstGreekAbbrCompletion(const std::string &s)
+{
+    std::string ret;
+    size_t sp = s.find_first_of(' ');
+    if (sp == std::string::npos)
+    {
+        int i = findGreekNameIndexBySubstr(s);
+        return (i >= 0) ? Greek::getInstance()->abbrevs[i] : s;
+    }
+    else
+    {
+        std::string prefix = s.substr(0, sp);
+        std::string suffix = s.substr(sp + 1);
```

Comment:
remove this line (https://app.codacy.com/app/375gnu/Celestia/pullRequest?prid=3158010 - it's actually useful :))